File: Python/4 - registro_calificaciones.py
def obtener_calificaciones():
    while True:
        try:
            num_calificaciones = int(input("Ingrese el numero de calificaciones que desea registrar: "))
            if num_calificaciones <= 0:
                raise ValueError("El numero de calificaciones debe ser mayor a cero.")
            return num_calificaciones
        except ValueError as e:
            print(f"Error: {e}. Intente de nuevo.")

def obtener_datos_estudiantes(num_estudiantes, num_calificaciones):
    estudiantes = []
    for i in range(num_estudiantes):
        while True:
            try:
                cedula = input(f"Ingrese la cedula del estudiante {i + 1}: ").strip()
                if not cedula:
                    raise ValueError("La cedula no puede estar vacía.")
                calificaciones = []
                for j in range(num_calificaciones):
                    while True:
                        try:
                            calificacion = float(input(f"Ingrese la calificacion {j + 1} para el estudiante con cedula {cedula}: "))
                            if calificacion < 0 or calificacion > 100:
                                raise ValueError("La calificacion debe estar entre 0 y 100.")
                            calificaciones.append(calificacion)
                            break
                        except ValueError as e:
                            print(f"Error: {e} Intente de nuevo.")
                estudiantes.append((cedula, calificaciones))
                break
            except ValueError as e:
                print(f"Error: {e}. Intente de nuevo.")
    return estudiantes

def guardar_calificaciones(estudiantes):
    for cedula, calificaciones in estudiantes:
        filename = f"{cedula}.txt"
        try:
            with open(filename, 'w') as file:
                for calificacion in calificaciones:
                    file.write(f"{calificacion}\n")
            print(f"Calificaciones del estudiante con cedula {cedula} guardadas en {filename}.")
        except IOError as e:
            print(f"Error al escribir el archivo {filename}: {e}")

while True:
    try:
        num_estudiantes = int(input("Ingrese el nmero de estudiantes: "))
        if num_estudiantes <= 0:
            raise ValueError("El numero de estudiantes debe ser mayor a cero.")
        break
    except ValueError as e:
        print(f"Error: {e}. Intente de nuevo.")

num_calificaciones = obtener_calificaciones()
estudiantes = obtener_datos_estudiantes(num_estudiantes, num_calificaciones)
guardar_calificaciones(estudiantes)

File: Julia/4 - registro_calificaciones.jl
function obtener_calificaciones()
    while true
        try
            println("Ingrese el numero de calificaciones que desea registrar: ")
            num_calificaciones = parse(Int16, readline())
            if num_calificaciones <= 0
                throw(ArgumentError("El numero de calificaciones debe ser mayor a cero."))
            end
            return num_calificaciones
        catch e
            println("Error: $e. Intente de nuevo.")
        end
    end
end

function obtener_datos_estudiantes(num_estudiantes, num_calificaciones)
    estudiantes = []

    for i in 1:num_estudiantes
        while true
            try
                println("Ingrese la cedula del estudiante $i: ")
                cedula = strip(readline())
                if isempty(cedula)
                    throw(ArgumentError("La cedula no puede estar vacía."))
                end

                calificaciones = []
                for j in 1:num_calificaciones
                    while true
                        try
                            println("Ingrese la calificacion $j para el estudiante con cedula $cedula: ")
                            calificacion = parse(Float64, readline())
                            if calificacion < 0 || calificacion > 100
                                throw(ArgumentError("La calificacion debe estar entre 0 y 100."))
                            end
                            push!(calificaciones, calificacion)
                            break
                        catch e
                            println("Error: $e Intente de nuevo.")
                        end
                    end
                end

                push!(estudiantes, (cedula, calificaciones))
                break
            catch e
                println("Error: $e. Intente de nuevo.")
            end
        end
    end

    return estudiantes
end

function guardar_calificaciones(estudiantes)
    for (cedula, calificaciones) in estudiantes
        filename = "$cedula.txt"
        try
            open(filename, "w") do file
                for calificacion in calificaciones
                    write(file, "$calificacion\n")
                end
            end
            println("Calificaciones del estudiante con cedula $cedula guardadas en $filename.")
        catch e
            println("Error al escribir el archivo $filename: $e")
        end
    end
end

function obtener_cantidad_estudiantes()
    while true
        try
            println("Ingrese el numero de estudiantes: ")
            num_estudiantes = parse(Int, readline())
            if num_estudiantes <= 0
                throw(ArgumentError("El numero de estudiantes debe ser mayor a cero."))
            end
            return num_estudiantes
        catch e
            println("Error: $e. Intente de nuevo.")
        end
    end
end

num_estudiantes = obtener_cantidad_estudiantes()
num_calificaciones = obtener_calificaciones()
estudiantes = obtener_datos_estudiantes(num_estudiantes, num_calificaciones)
guardar_calificaciones(estudiantes)